glean-core/android/src/main/java/mozilla/telemetry/glean/scheduler/PingUploadWorker.kt

```diff
@@ -62,8 +63,11 @@ class PingUploadWorker(context: Context, params: WorkerParameters) : Worker(cont
          * @return true if process was successful
          */
         internal fun uploadPings(): Boolean {
-            val httpPingUploader = HttpPingUploader()
-            return httpPingUploader.process()
+            if (Glean.getUploadEnabled()) {
```

Comment:
There was a corresponding change when this "clearing metrics" thing was added to [glean-ac](https://github.com/mozilla-mobile/android-components/pull/2838).  There has always been a check to not collect and assemble a ping when telemetry is disabled.  This just prevents any already queued ones from being sent.  But even then, any queued pings should be deleted from disk when disabling telemetry before we even get here, so it's kind of like an extra insurance policy.

There is a test (in this PR) to check that when telemetry is disabled that the queued pings are cleared.

However, I don't think there's anything that explicitly tests this block -- it will be a bit hard to do, because you'd have to disable telemetry without clearing the queued pings, which we can't really do from the Kotlin side...  I'll have to think on how to test that...

Reply:
Nah, thanks for clarifying. No need to test this specifically, not in this PR at least. On the nit side, what if this becomes:

```
return if (Glean.getUploadEnabled()) {
  HttpPingUploader().process()
} else {
  false
}
```

It should prevent a Kotlin warning.